test/ctia/test_helpers/core.clj

```diff
@@ -47,44 +51,56 @@
   (binding [*config-transformers* (conj *config-transformers* tf)]
     (body-fn)))
 
-(def with-properties-vec
-  (mth/build-with-properties-vec-fn PropertiesSchema))
+(s/defn with-properties*
+  [properties-vec :- (s/pred vector?)
+   f :- (s/=> s/Any)]
+  (assert (not (thread-bound? #'*current-app*))
+          "Cannot override properties after TK app has started!")
+  (assert (even? (count properties-vec))
+          (str "Even count required for properties-vec, actual " (count properties-vec)))
+  (binding [*properties-overrides* (into *properties-overrides* properties-vec)]
+    (f)))
 
-(defmacro with-properties [properties-vec & sexprs]
-  `(with-properties-vec ~properties-vec
+(defmacro with-properties
+  "Simulates setting the specified System properties to configure
+  the creation of a TK app, except thread-locally.
```

Comment:
`except` -> `except that it is` or so

(else one could think you're saying the opposite)